### PR TITLE
removed python 2 support

### LINKED ISSUE #1120: Drop Python 2 support
~~marshmallow 3 will be the last major version to support Python 2.~~ Per discussion below, Python 2 support will be dropped in marshmallow 3.

### PR DESCRIPTION
**Pull request by rooterkyberian**
resolves #1120

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -65,13 +65,11 @@ def read(fname):
         'serialization', 'rest', 'json', 'api', 'marshal',
         'marshalling', 'deserialization', 'validation', 'schema',
     ],
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
+    python_requires='>=3, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*',
```
Comment (sloria) [inline setup.py]: Can't this be simplified to `>=3.5`?

File: tests/test_validate.py [hunk 2]
```diff
@@ -511,7 +510,7 @@ def test_predicate_repr():
     assert (
         repr(validate.Predicate(method='foo', error='bar', zoo=1)) ==
         '<Predicate(method={!r}, kwargs={!r}, error={!r})>'
-        .format('foo', {str('zoo') if PY2 else 'zoo': 1}, 'bar')
+        .format('foo', {'zoo': 1}, 'bar')
```
Comment (sloria) [inline tests/test_validate.py:513]: We'll probably want to switch to f-strings, but this is fine.

Comment (sloria) [reply tests/test_validate.py:513]: Oh scratch that. I forget that f-strings aren't in 3.5, which we're supporting.

**Comment by sloria** [comment]
Can you add the `--py3-plus` flag to the `pyupgrade` hook in `.pre-commit-config.yaml` and run it?

File: tests/test_validate.py (hunk 2 above)
Comment (rooterkyberian) [reply tests/test_validate.py:513]: 3.5 EoL in 2020-09-13 so it probably will be still a while

**Comment by lafrech** [comment]
AUTHORS.rst needs to be updated.

**Comment by sloria** [comment]
Thank you for the PR!

**Pull request merged**